### ISSUE TITLE
Design document for the new "life" epic

### ISSUE THREAD

**Issue by clarafu** [concourse/concourse#1029]
ref. #629

With all the new changes to volume and container lifecycle, it would be great if there could be a document outlining the new abstract model that is built. This would be very helpful for the community and to the team itself. 😊

**Comment by vito** [comment]
We started on a Gliffy diagram, but haven't integrated it anywhere.

We should do this in the concourse/concourse wiki and have that as part of the document.

**Comment by topherbullock** [comment]
*Here's a list of things which should be covered in doc(s):*
- Containers and why you should care about them
- How Containers are GC'd
- Volumes and why you should care about them
- How Volumes are GC'd

**Comment by vito** [comment]
We're thinking of having graphics for the following:

1. From a freshly deployed Concourse, configure a pipeline with a resource and job, run a build, and show a timeline of how containers/volumes/workers *appear*.
2. Show a diagram of the database schema relationships to show how containers/volumes *go away*.

The document can then serve to augment these (the *what*) with why we made the decisions we made (the *why*). For example, a large part of the schema's design is to prevent the destroying of containers/volumes that are in flight, and to utilize database integrity to this end.
